sklearn/preprocessing/tests/test_encoders.py

```diff
@@ -677,3 +677,10 @@ def test_categories(density, drop):
 @pytest.mark.parametrize('Encoder', [OneHotEncoder, OrdinalEncoder])
 def test_encoders_has_categorical_tags(Encoder):
     assert 'categorical' in Encoder()._get_tags()['X_types']
+
+
+def test_ordinal_encoder_deprecated_categories_auto_with_string_labels():
+    enc = OrdinalEncoder()
+    X = [['Male', 1], ['Female', 3], ['Female', 2]]
+    with pytest.warns(DeprecationWarning, match="From version 0.24"):
+        enc.fit(X)
```

Comment:
Please check that the fit is consistent with 'sort'

Reply:
Sure. Thanks for the inputs.